pkg/cnb/create_builder.go

```diff
@@ -110,6 +128,20 @@ func (r *RemoteBuilderCreator) CreateBuilder(ctx context.Context, builderKeychai
 	return builder, nil
 }
 
+func ResolveBuilderRef(ref, builderKind, builderName string) (string, error) {
```

Comment:
It seems that this is called by the consumer of CreateBuilder. Should it be in this package?

Reply:
I mainly put it here so I wouldn't need to duplicate the logic in both reconcilers

Reply:
Then should it be within the RemoteBuilderCreator?

Reply:
I'm fine with that

Reply:
this would require us to pass in the builder name and kind into createBuilder though

Reply:
which is fine imo

Reply:
mentioned the same above, but I ended up just moving the implementations into each reconciler since the ended up looking slightly different. The alternative i was thinking was passing an interface to create_builder that exposes the name, kind, and build spec (or just the specific fields from the build spec).